code/src/Models/User.ts

```diff
@@ -2,58 +2,90 @@ import app from "firebase"
 
 
 export default class User{
+    /**
+     * User class contains uid, firstName, lastName
+     * @author Mohamad Abdel Rida
+     * 
+     * 
+     */
+
+
     firstName?:string
     lastName?:string
+    email:string
     uid:string
-    // TODO add other attributes
-    constructor(uid:string,firstName?:string, lastName?:string){
+
+    /**
+     * 
+     * @param uid user id used in firebase
+     * @param firstName user's first name | from firestore or user input
+     * @param lastName user's last name | from firestore user input
+     * @param email user's email address
+     */
+
+    constructor(uid:string,email:string,firstName?:string, lastName?:string){
         this.firstName = firstName
         this.lastName =lastName
         this.uid = uid
+        this.email = email
     }
 
     private async addToDb(){
-        //Adds the user to the users colleciton 
-        
-        const doc = await app.firestore().collection('users').doc(this.uid).set(
+        /**
+         * method on User
+         * adds user fields to a unique document in firestore using uid
+         * 
+        */
+        try{
+            await app.firestore().collection('users').doc(this.uid).set(
             {
                 firstName:this.firstName,
                 lastName:this.lastName,
-                //TOOD add other fields
+               
             }
-        )
+        )}catch(e){
+            console.log(e)
+        }
     
     }
-    async fetchUserDetails(){
-        //get's user information from firebase
-        // returns Promise with User model if transaction completed
-        // TODO Add props to list and iterate instead of getting each 
-        // field line by line?
+    private async fetchUserDetails(){
```

Comment:
For linting purposes it might be best to explicitly state what type of object is being returned it. You might also want to consider creating an interface for this data so it can be a bit more strongly typed